Example of Thread creation in documentation does not compile

**Describe the bug**
Creating a statically defined thread as mentioned in the [Thread ](https://docs.zephyrproject.org/latest/reference/kernel/threads/index.html#spawning-a-thread)documentation fails to compile. 

Changing `K_NO_WAIT` to `0` fixes the problem. 

**To Reproduce**
Steps to reproduce the behavior:
1. Create thread using `K_THREAD_DEFINE(my_tid, MY_STACK_SIZE,my_entry_point, NULL, NULL, NULL,MY_PRIORITY, 0, K_NO_WAIT);`
2. Compile
3. See error

**Expected behavior**
Example should compile.

**Impact**
Not much, just a bit of time.

**Screenshots or console output**
```
C:/zephyrproject/zephyr/include/kernel.h:1078:16: error: incompatible types when initializing type 'int' using type 'k_timeout_t' {aka 'struct <anonymous>'}
 1078 |  .init_delay = (delay),                                   \
      |                ^
C:/zephyrproject/zephyr/include/kernel.h:1122:3: note: in expansion of macro 'Z_THREAD_INITIALIZER'
 1122 |   Z_THREAD_INITIALIZER(&_k_thread_obj_##name,   \
      |   ^~~~~~~~~~~~~~~~~~~~
C:/zephyrproject/zephyr/samples/hello_world/src/main.c:19:1: note: in expansion of macro 'K_THREAD_DEFINE'
   19 | K_THREAD_DEFINE(my_tid, MY_STACK_SIZE,
      | ^~~~~~~~~~~~~~~
```

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain GCC
 - master

--- a/doc/reference/kernel/threads/index.rst
+++ b/doc/reference/kernel/threads/index.rst
@@ -341,7 +341,13 @@ The following code has the same effect as the code segment above.
 
     K_THREAD_DEFINE(my_tid, MY_STACK_SIZE,
                     my_entry_point, NULL, NULL, NULL,
-                    MY_PRIORITY, 0, K_NO_WAIT);
+                    MY_PRIORITY, 0, 0);
+
+.. note::
+   The delay parameter to :cpp:func:`k_thread_create()` is a
+   :c:type:`k_timeout_t` value, so :c:macro:`K_NO_WAIT` means to start the
+   thread immediately. The corresponding parameter to :c:macro:`K_THREAD_DEFINE`
+   is a duration in integral milliseconds, so the equivalent argument is 0.
 
 User Mode Constraints
 ---------------------

